signer/src/stacks/contracts.rs

```diff
@@ -987,11 +1066,118 @@ impl AsContractCall for RejectWithdrawalV1 {
     ///    an event on the canonical Stacks blockchain.
     /// 2. That the signer bitmap matches the signer decisions stored in
     ///    this signer's database.
-    async fn validate<C>(&self, _ctx: &C, _req_ctx: &ReqContext) -> Result<(), Error>
+    async fn validate<C>(&self, ctx: &C, req_ctx: &ReqContext) -> Result<(), Error>
     where
         C: Context + Send + Sync,
     {
-        // TODO(255): Add validation implementation
+        eprintln!("check deployer");
+        if self.deployer != req_ctx.deployer {
+            return Err(WithdrawalRejectErrorMsg::DeployerMismatch.into_error(req_ctx, self));
+        }
+
+        // 1. The request exists. Check whether the associated withdrawal request transaction is confirmed on the canonical stacks blockchain. Fail the withdrawal request if it is not on the canonical stacks blockchain.
+        //
+        // 2. The double spend check. Check whether the qualified request ID is in the bitcoin_withdrawals_outputs table, and that any of the associated txids are confirmed on the canonical bitcoin blockchain. Fail the withdrawal request if such a transaction was found.
+
+        eprintln!("get stacks chain tip");
+        let stacks_chain_tip = ctx
+            .get_storage()
+            .get_stacks_chain_tip(&req_ctx.chain_tip.block_hash)
+            .await?;
+        let Some(stacks_block) = stacks_chain_tip else {
+            eprintln!("no stacks block for request chain tip");
+            return Err(WithdrawalRejectErrorMsg::MissingStacksBlock.into_error(req_ctx, self));
+        };
+        let stacks_block_hash = stacks_block.block_hash;
+
+        eprintln!("get withdrawal signers");
+        let withdrawal_signers = &ctx
+            .get_storage()
+            .get_withdrawal_signers(self.request_id, &stacks_block_hash)
```

Comment:
Ahh, yeah one thing that needs to change here is the `RejectWithdrawalV1` itself. The `request_id` should be a `QualifiedRequestId` instead of a `u64` and we should probably rename the field too. Then you could use the block hash in the qualified ID here to get the votes.

Reply:
Elsewhere in the code it looks like we name `QualifiedRequestId` fields simply as `id`, and accessor functions `qualified_id()`.  What name would you like to see?

Reply:
I tried `qualified_request_id` and it's really verbose.  Any thoughts on `qualified_id` or just `id`?

Reply:
I'd probably go with `qualified_id`.

Reply:
I went with `id`, since other code did so.  It seems like the goal was to avoid namespace clashes with the `qualified_id()` accessors.  I fixed it in the code, the protobufs, and the conversions.